src/utils/spline_builder_deriv_field_2d.hpp

```diff
@@ -0,0 +1,345 @@
+        ddc::BoundCond BoundCond1max,
+        ddc::BoundCond BoundCond2min,
+        ddc::BoundCond BoundCond2max,
+        class LayoutSpline = Kokkos::layout_right,
+        class LayoutFunction = Kokkos::layout_right>
+class SplineBuliderDerivField2D
+{
+    using MemorySpace = typename ExecSpace::memory_space;
+
+    using Dim1 = typename BSplines1::continuous_dimension_type;
+    using Dim2 = typename BSplines2::continuous_dimension_type;
+    using Deriv1 = ddc::Deriv<Dim1>;
+    using Deriv2 = ddc::Deriv<Dim2>;
+
+    using Builder2D = ddc::SplineBuilder2D<
+            ExecSpace,
+            MemorySpace,
+            BSplines1,
+            BSplines2,
+            Grid1,
+            Grid2,
+            BoundCond1min,
+            BoundCond1max,
+            BoundCond2min,
+            BoundCond2max,
+            ddc::SplineSolver::LAPACK>;
+
+    using SplineType = DField<IdxRange<BSplines1, BSplines2>, MemorySpace, LayoutSpline>;
+    using DerivFieldType = DerivField<
+            double,
+            IdxRange<Deriv1, Grid1, Deriv2, Grid2>,
+            MemorySpace,
+            LayoutFunction>;
+
+    using FunctFieldMem = DFieldMem<IdxRange<Grid1, Grid2>, MemorySpace>;
+    using FunctField = DField<IdxRange<Grid1, Grid2>, MemorySpace>;
+
+    using CrossDerivFieldMem = DFieldMem<IdxRange<Deriv1, Deriv2>, MemorySpace>;
+    using CrossDerivField = DField<IdxRange<Deriv1, Deriv2>, MemorySpace>;
+    using CrossDerivConstField = DConstField<IdxRange<Deriv1, Deriv2>, MemorySpace>;
+
+    using Deriv1FieldMem = DFieldMem<IdxRange<Deriv1, Grid2>, MemorySpace>;
+    using Deriv1Field = DField<IdxRange<Deriv1, Grid2>, MemorySpace>;
+    using Deriv1ConstField = DConstField<IdxRange<Deriv1, Grid2>, MemorySpace>;
+
+    using Deriv2FieldMem = DFieldMem<IdxRange<Grid1, Deriv2>, MemorySpace>;
+    using Deriv2Field = DField<IdxRange<Grid1, Deriv2>, MemorySpace>;
+    using Deriv2ConstField = DConstField<IdxRange<Grid1, Deriv2>, MemorySpace>;
+
+private:
+    Builder2D const& m_builder;
+
+public:
+    /**
+     * @brief Instantiate the class by storing a reference to the spline builder 
+     * we can to use. 
+     * 
+     * @param[in] builder A reference to SplineBuilder2D from DDC that we store
+     * in the class. 
+     */
+    explicit SplineBuliderDerivField2D(Builder2D const& builder) : m_builder(builder) {}
+
+    /**
+     * @brief Build the spline representation of the function given in a 
+     * DerivField applying the referenced spline builder stored in the class. 
+     * 
+     * @param[out] spline Spline coefficients on a 2D grid. 
+     * @param[in] function_and_derivs Data defining the function on a 2D grid. 
+     */
+    void operator()(SplineType spline, DerivFieldType function_and_derivs) const
+    {
+        // Get the fields on the right layout.
+        // --- get the index ranges.
+        IdxRange<Grid1, Grid2> idx_range = get_idx_range(function_and_derivs.get_values_field());
```

Comment:
It is possible to do:
```suggestion
        IdxRange<Grid1, Grid2> idx_range = get_idx_range(function_and_derivs);
```
But feel free to keep the version that you find cleaner

Reply:
https://github.com/gyselax/gyselalibxx/pull/475/commits/6ea96f91d04f83f9edc1b2fc8c5f4d7b6e8fa191